### PR TITLE
Fix all `i18n` issues detected by new script

### LINKED ISSUE #9486: Use new `pre-commit` script to detect and fix existing `i18n` syntax issues
### A clear and concise description of what you want to happen
The new `detect_missing_i18n` script by @pidgezero-one has given us a super helpful list of `i18n` syntaxing issues throughout the codebase and/or valid `i18n` that accidentally gets flagged by the script.

These files are listed in `EXCLUDE_LIST` in `detect_missing_i18n.py`.

### What is the proposed solution / implementation?
Once the PR (#9248) is merged, I propose going through the full list of excluded files myself and making any changes where needed, either fixing `i18n` syntax or adjusting distribution of text to avoid accidental flags of the script. I'll also make any updates to the `i18n` Developer's Guide where needed as a result of edge-cases I discover in the process. 

### Related files

`detect_missing_i18n.py`

### Stakeholders

@pidgezero-one 

<hr>

### Instructions for Contributors

- Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

### PR DESCRIPTION
**Pull request by rebecca-shoptaw**
Closes #9486.

Fix. Manually fixes all `i18n` syntax issues picked up by the new `pre-commit` script. Also escapes some non-`i18n`-able and/or accidentally detected lines.

### Technical

- Involved adjusting the warn regex so that it detects `$('` or `$("` to avoid creating `WARN`s for Python code inserts
- I'm now seeing a `SyntaxWarning` when the new `messages.pot` is generated; would be nice to get this fixed before merge, but everything still runs as expected
- Had to manually escape some Python inserts that began with quotes i.e. `$(" " if condition else None)` --> `$((" " if condition else None))` -- somewhat awkward but only done in ~2-3 infrequently used files, so seems to be worth it
- There are also two very interesting edge-cases that I left in the `EXCLUDE_FILES` for now -- `/books/edit.html` and `/history/sources.html` -- as both have HTML elements concatenated with `+`:
https://github.com/internetarchive/openlibrary/blob/3a59fc5b9ae13876d98b916af87fcfc57b885cfa/openlibrary/templates/books/edit.html#L30
I'm not sure the best course of action for handling this would be (i.e. reverting to normal HTML, leaving them in the `EXCLUDE_LIST`, using the new escape syntax, something else, etc.)

### Testing

1. `pre-commit run --all-files` with the current much shorter exclude list
2. You should see several warnings, all intentional to indicate purposefully escaped text, but the test should pass

### Stakeholders

@cdrini @pidgezero-one

### CODE REVIEW

**Comment by rebecca-shoptaw** [comment]
> That's an interesting edge case - just my two cents, if concatenating HTML strings in that manner is infrequent/something we tend to avoid, I think it would be fine to just flag it with `$# detect-missing-i18n-skip-line`!

@pidgezero-one I figure this will probably be the best solution, just a bit awkward for the files themselves as they each have many instances of the concatenated HTML, which will require many flags. But we'll see!

**Comment by pidgezero-one** [comment]
@rebecca-shoptaw Ahh, I see - what about ignoring any tags followed by `' + `, or is that not universal enough?

**Comment by pidgezero-one** [comment]
@rebecca-shoptaw Ahh, I see... It might be a little tricky, but I wonder if it's worth it to add some logic that detects if it's part of a `$code` block?

I guess HTML within `$code` blocks might still need i18n, so maybe even just ignoring tags that are followed by `' + _` - but I wonder if this isn't universal enough!

File: openlibrary/macros/IABook.html [hunk 1]
```diff
@@ -19,7 +19,7 @@ <h3 itemprop="name" class="booktitle">
            </h3>
         </div>
     </div>
-    $# The following (empty) <div> is to aid in formatting the layout.
```
Comment (mekarpeles) [inline openlibrary/macros/IABook.html:22]: The benefit of `$#` is that this is visible to developers in the code, but it doesn't create a comment that needs to then be wastefully sent over the wire to patrons (extra kb's).

File: openlibrary/templates/search/work_search_facets.html [hunk 2]
```diff
@@ -16,15 +16,15 @@
...
-            "language": "Language",
+            "has_fulltext": _("Ebook"),
+            "author_key": _("Author"),
+            "subject_facet": _("Subjects"),
+            "person_facet": _("People"),
+            "place_facet": _("Places"),
+            "time_facet": _("Times"),
+            "first_publish_year": _("FirstPublished"),
+            "publisher_facet": _("Publisher"),
+            "language": _("Language"),
```
Comment (mekarpeles) [inline openlibrary/templates/search/work_search_facets.html:27]: We'll want to confirm that, in cases where we have dictionaries like this, that the values aren't being used as `keys` elsewhere in the code -- otherwise, once we i18n these keys, we could unintentionally break things. I don't think that's what's happening here, just sharing a possible use case we may hit given this PR touches some 100 files

Comment (rebecca-shoptaw) [reply openlibrary/templates/search/work_search_facets.html:27]: Roger that! I'll investigate -- and I believe this is the only dict affected by the PR (the script doesn't actually detect them, I just ran across it).

File: openlibrary/macros/IABook.html (hunk 1 above)
Comment (rebecca-shoptaw) [reply openlibrary/macros/IABook.html:22]: Aha! I'll use the `$# detect-missing-i18n-skip-line` to handle these instead

**Comment by rebecca-shoptaw** [comment]
Per @cdrini's advice, closing this to split it up into easier-to-review sub-PRs, which I will list and categorize in the original issue. 🙂
